quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -226,6 +227,23 @@ instance IsTest QC where
       (qcOutputNl ++
         (if putReplayInDesc then replayMsg else ""))
 
+
+-- | Like the original 'QC.quickCheck' but is reporting progress using tasty
+-- callback.
+--
+quickCheck :: (Progress -> IO ())
+           -> QC.Args
+           -> QC.Property
+           -> IO QC.Result
+quickCheck yieldProgress args prop = do
+  -- Here we rely on the fact that QuickCheck currently prints its progress to
+  -- stderr and the overall status (which we don't need) to stdout
+  tm <- QC.newTerminal
```

Comment:
I'm somewhat uneasy with depending on undocumented internal parts of QuickCheck: these entities are deliverately hidden from Haddocks. But I cannot suggest any better approach.